pkg/plugin/local/local.go

```diff
@@ -85,10 +85,43 @@ func (lds *LocalDataSource) AddTrustZone(trustZone *trust_zone_proto.TrustZone)
 	return trustZone, nil
 }
 
-func (lds *LocalDataSource) GetTrustZone(id string) (*trust_zone_proto.TrustZone, error) {
-	trustZone, ok := lds.config.GetTrustZoneByName(id)
+func (lds *LocalDataSource) DestroyTrustZone(name string) error {
+	// Fail if any clusters exist in the trust zone.
+	if len(lds.config.GetClustersByTrustZone(name)) > 0 {
+		return fmt.Errorf("one or more clusters exist in trust zone %s in local config", name)
+	}
+	// Deleting the trust zone also implicitly removes any attestation policies and federations
+	// bound to it because they are stored in the trust zone message.
+	// Federations in other trust zones that reference this trust zone need to be cleaned up.
```

Comment:
Not for this PR obviously but do we have a feel for how this garbage collection might happen? Could be tricky once the federation landscape gets busy

Reply:
It's already happening in the loop below - see slices.DeleteFunc